scipy/linalg/special_matrices.py

```diff
@@ -1190,3 +1190,121 @@ def fiedler_companion(a):
     c[[0, 1], 0] = [-a[1], 1]
 
     return c
+
+
+def convolution_matrix(a, n, mode='full'):
+    """
+    Construct a convolution matrix.
+
+    Constructs the dense matrix representing one-dimensional
+    convolution [1]_.
+
+    ``A = convolution_matrix(a, n[, mode])`` creates a matrix ``A`` such
+    that ``A @ v`` (or ``matmul(A, v)``) is equivalent to using
+    ``convolve(a, v[, mode])``. In the default 'full' mode, the entries of
+    ``A`` are given by ``A[i, j] == (a[i-j] if (0 <= (i-j) < m) else 0)``.
+
+    Parameters
+    ----------
+    a : (m,) array_like
+        The 1-D array to convolve.
+    n : int
+        The number of columns in the resulting matrix.
+        This is analogous to the length of `v` in ``numpy.convolve(a, v)``.
+    mode : str
+        This is analogous to `mode` in numpy.convolve(v, a, mode).
+        It must be one of ('full', 'valid', 'same').
+        See below for how `mode` determines the shape of the result.
```

Comment:
double backticks around `mode`

Reply:
For years, we used single backticks around single names, and in fact, the NumPy docstring still says to use them, in a couple places.  E.g. from https://numpydoc.readthedocs.io/en/latest/format.html#sections, scroll down to the section "4. Parameters", where it says

> Enclose variables in single backticks.

And later, in https://numpydoc.readthedocs.io/en/latest/format.html#common-rest-concepts,

> Variable, module, function, and class names should be written between single back-ticks (\`numpy\`).

Reply:
I'm sure I've been in a conversation about the question of single or double backticks around variable names some time in the past, but now I can't find it.

Reply:
Single ones resolve to internal cross references as far as I know. But yes this is confusing now for me.